distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Postgres/Postgres_Dialect.enso

```diff
@@ -486,6 +486,14 @@ make_contains_expr expr substring =
 ## PRIVATE
 contains = Base_Generator.lift_binary_op "contains" make_contains_expr
 
+## PRIVATE
+left = Base_Generator.lift_binary_op "LEFT" str-> n->
+    Builder.code "left(" ++ str ++ ", CAST(" ++ n ++ " AS INT))"
```

Comment:
why is this cast needed?

Reply:
That's a good question. I was getting a bigInt not supported error message without it, but I meant to go back and find out why it thinks n is a bigInt. I will do that.

Reply:
2024-01-11 13:41:36 2024-01-11 13:41:36.905 UTC [68] LOG:  execute <unnamed>: SELECT left("table_4-a1e769cc-6154-4b73-96d1-d55f78b41fc8"."strings", CAST($1 AS INT)) AS "text_left([strings], 1)" FROM "table_4-a1e769cc-6154-4b73-96d1-d55f78b41fc8" AS "table_4-a1e769cc-6154-4b73-96d1-d55f78b41fc8"
2024-01-11 13:41:36 2024-01-11 13:41:36.905 UTC [68] DETAIL:  parameters: $1 = '1'

I'm thinking the single quotes around the 1 in the parameter definition are the problem

Reply:
Curious.

I'm pretty sure the numeric constant should be sent as integer, definitely not text.

---

I think I know what may be the issue:
https://github.com/enso-org/enso/blob/develop/distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Statement_Setter.enso#L35-L43

Here we have the logic for setting parameters. If it fits in `long` it is set as `long`, which is `int8 (bigint)` in Postgres, if it is larger, it is set as `BigDecimal`.

However here it still is too large - because Postgres expects `int4`.

I see 2 solutions:
1. keep the cast then
2. change the logic, to send smaller integer constants in a smaller type (e.g. `int4`) if they fit it.

The (2) would have broader implications. I think we might think about it, but I'm not sure if we want the type to vary depending on the value - I imagine in some rare cases it could make an expression work or break depending on the magnitude of the value - which I'm not sure is correct.

The cast will also ensure that even if we provide a column for `n` and that column is a BigInteger column (which passes the `Value_Type.expect_numeric` check), it will still work OK as long as the values are small enough.

After that, I'm convinced that the CAST seems OK here.